src/allmydata/node.py

```diff
@@ -203,7 +203,13 @@ def config_from_string(basedir, portnumfile, config_str):
     # load configuration from in-memory string
     parser = ConfigParser.SafeConfigParser()
     parser.readfp(BytesIO(config_str))
-    return _Config(parser, portnumfile, basedir, '<in-memory>')
+
+    def write_new_config(cfg):
+        """
+        We throw away any attempt to persist
+        """
+        pass
```

Comment:
I don't think it can be (or at least, not trivially). It's used by a bunch of tests .. I tried raising an error here but that causes at least some `test_connections` tests to fail.

Longer-term I'd hope we can further abstract "configuration stuff" and then maybe we could have an in-memory-only test thing (that *can* persist).